metals/src/main/scala/scala/meta/internal/debug/DebugSession.scala

```diff
@@ -0,0 +1,68 @@
+package scala.meta.internal.debug
+
+import java.io.File
+import java.net.URI
+import java.nio.file.Files
+import java.nio.file.Files.createTempDirectory
+import java.nio.file.Paths
+import org.eclipse.lsp4j.debug.services.IDebugProtocolClient
+import scala.concurrent.ExecutionContext
+import scala.concurrent.Future
+import scala.meta.internal.debug.protocol.LaunchParameters
+import scala.meta.internal.debug.protocol.OutputEventArguments._
+import scala.sys.process.Process
+import scala.sys.process.ProcessLogger
+
+final class DebugSession(
+    process: Process,
+    val exitCode: Future[Int]
+) {
+  def whenTerminated[A](f: Int => A)(implicit ec: ExecutionContext): Future[A] =
+    exitCode.map(f)
+}
+
+final class DebugSessionFactory(client: IDebugProtocolClient) {
+  def create(
+      params: LaunchParameters
+  )(implicit ec: ExecutionContext): DebugSession = {
+    val command = createCommand(params)
+    val workingDir = createWorkingDir(params)
+    val listener = new Listener()
+
+    val process = Process(command, workingDir).run(listener)
```

Comment:
What do you think about using `NuProcess` here? I've found `java.lang.ProcessBuilder` and `scala.sys.Process` to be inadequate for a many reasons.

Reply:
Oh, I did not notice it was in scope. But for now let's leave it as it is (even though `NuProcess` is much better to use) - I will revisit this idea if my approach of using a build server to run processes fails, because if it succeeds then there will be no need for using processes directly anyway

Reply:
I think it would be good to try to reach the final design (BSP run vs. NuProcess) in this PR before it's merged.

Reply:
Did you consider using the `runInTerminal` request in DAP? https://microsoft.github.io/debug-adapter-protocol/specification#Reverse_Requests_RunInTerminal

Reply:
> I think it would be good to try to reach the final design (BSP run vs. NuProcess) in this PR before it's merged.

We would depend on at least `BSP` v. 2.0.0-M4. `bloop` is using 2.0.0-M1, therefore either we postpone this feature or use `NuProcess`

Reply:
Changed implementation to use a `NuProcess`